litestar/utils/dataclass.py

```diff
@@ -91,15 +97,18 @@ def simple_asdict(obj: DataclassProtocol, exclude_none: bool = False, exclude_em
         obj: A dataclass instance.
         exclude_none: Whether to exclude None values.
         exclude_empty: Whether to exclude Empty values.
+        by_alias: Whether to use alias for the field name.
+        exclude: Name of attributes to be excluded.
 
     Returns:
         A dictionary of key/value pairs.
     """
     ret = {}
-    for field in extract_dataclass_fields(obj, exclude_none, exclude_empty):
+    for field in extract_dataclass_fields(obj, exclude_none, exclude_empty, exclude=exclude):
+        field_name = field.name if not by_alias else "-".join(field.name.split("_"))
```

Comment:
I don't think `by_alias` belongs in this method, at least with this implementation, as this seems like functionality specific to the headers use-case, and dataclass fields don't really have the concept of a serialization alias.